src/cl_wrapper/cl_interceptor.c

```diff
@@ -642,24 +657,25 @@ clCreateBuffer(cl_context   context,
         }
         else
         {
-            fill_ptr = (char*)calloc(sizeof(char), size + POISON_FILL_LENGTH);
+            fill_ptr = (char*)calloc(sizeof(char), size + 2*POISON_FILL_LENGTH);
```

Comment:
I'm not entirely sure I like just randomly having around a "2*" with no reason for why it's like this. What if I want to *not* do underflow detection (e.g. I'm on a system where the whole SubBuffer idea doesn't work)? Could we either make this a compile-time constant or a global variable that's descriptive of what the "2" stands for?

Reply:
Note that this suggestion holds for basically every use of "2*" throughout all the rest of the diffs.